src/dune_rules/melange_rules.ml

```diff
@@ -43,7 +43,7 @@ let js_includes ~sctx ~target_dir ~requires_link ~scope ~js_ext =
              ; Hidden_deps hidden_deps
              ])))
 
-let build_js ~loc ~dir ~pkg_name ~module_system ~dst_dir ~obj_dir ~sctx
+let build_js ~loc ~dir ~pkg_name ~mode ~module_system ~dst_dir ~obj_dir ~sctx
```

Comment:
Maybe out of scope, but there are already 4 params that depend on `mel` value: `loc`, `mode`, `module_system` and `js_ext`. I wonder at what point it's just easier to pass the `mel` value directly.

Reply:
Even though it's easier to pass more than what we need, we pay for that by making it harder to understand the function.